service/storage_service.cc

```diff
@@ -311,6 +311,15 @@ future<> storage_service::topology_state_load(cdc::generation_service& cdc_gen_s
     // read topology state from disk and recreate token_metadata from it
     _topology_state_machine._topology = co_await _sys_ks.local().load_topology_state();
 
+    co_await _feature_service.container().invoke_on_all([&] (gms::feature_service& fs) {
+        return fs.enable(boost::copy_range<std::set<std::string_view>>(_topology_state_machine._topology.enabled_features));
+    });
+
+    // Update the legacy `enabled_features` key in `system.scylla_local`.
+    // It's OK to update it after enabling features because `system.topology` now
+    // is the source of truth about enabled features.
+    co_await _sys_ks.local().save_local_enabled_features(_topology_state_machine._topology.enabled_features);
```

Comment:
nit: It's perfectly worth it to make a static method for `persistent_feature_enabler` and call it here and in `persistent_feature_enabler::enable_features()`. It's not a review blocker, of course.